source/appModules/devenv.py

```diff
@@ -82,6 +87,41 @@ def _get_DTE(self):
 		return DTE
 
 
+class VsWpfTextViewTextInfo(UIATextInfo):
+
+	def _getLineNumberString(self, textRange):
+		# Visual Studio exposes line numbers as part of the actual text.
+		# We want to store the line number in a format field instead.
+		lineNumberRange = textRange.Clone()
+		lineNumberRange.MoveEndpointByRange(
+			UIAHandler.TextPatternRangeEndpoint_End,
+			lineNumberRange,
+			UIAHandler.TextPatternRangeEndpoint_Start
+		)
+		return lineNumberRange.GetText(-1)
+
+	def _getFormatFieldAtRange(self, textRange, formatConfig, ignoreMixedValues=False):
+		formatField = super()._getFormatFieldAtRange(textRange, formatConfig, ignoreMixedValues=ignoreMixedValues)
+		if not formatField or not formatConfig['reportLineNumber']:
+			return formatField
+		lineNumberStr = self._getLineNumberString(textRange)
+		if lineNumberStr:
+			try:
+				formatField.field['line-number'] = int(lineNumberStr)
+			except ValueError:
+				pass

```

Comment:
should this log something? in what cases is lineNumberStr not an integer?

Reply:
I don't expect this to fail, though it certainly makes sense to log something.